examples/eks_test_fixture/main.tf

```diff
@@ -48,10 +48,11 @@ module "vpc" {
 }
 
 module "eks" {
-  source        = "../.."
-  cluster_name  = "${local.cluster_name}"
-  subnets       = "${module.vpc.public_subnets}"
-  tags          = "${local.tags}"
-  vpc_id        = "${module.vpc.vpc_id}"
-  worker_groups = "${local.worker_groups}"
+  source         = "../.."
+  cluster_name   = "${local.cluster_name}"
+  subnets        = ["${module.vpc.public_subnets}", "${module.vpc.private_subnets}"]
+  worker_subnets = "${module.vpc.private_subnets}"
```

Comment:
This can actually still be done using the above, `worker_subnets` is just a list so you could, for example, have:
`worker_subnets = ["var.my_public_subnet", "var.my_private_subnet", "subnet-abcde123"]` and all of these subnets would be used in the autoscaling group to place workers into.

Reply:
Hmm maybe I haven't explained myself well enough... Consider:

```
locals {
  worker_groups = "${list(
                  map("instance_type","t2.small",
                      "additional_userdata","echo foo bar",
                      "name", "foo",
                   ),
                  map("name", "bar",
                   ),
  )}"
```

The above arrangement of `worker_groups` gives me 2 distinct ASGs with workers of different size, name, and userdata, but with the code in your branch, there'd be no way to put `foo` ASG into strictly the public subnets while putting `bar` in strictly private subnets. The `worker_subnets` variable is currently shared amongst each of _n_  autoscaling groups if it's not split into a key/value like `name`, `instance_type`, etc. Does that make any better sense? Help me come around with an example if I'm missing something.

Reply:
My option for this is, there should be no kubernetes worker nodes put in public subnets at all.  

Only public facing load balancers, Nat gateway, WAF (application firewalls), bastion/jump box or similars should be put in public subnets.